admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/SecurityConfig.kt

```diff
@@ -33,26 +34,46 @@ import io.ktor.client.call.*
 import io.ktor.client.engine.apache.*
 import io.ktor.client.request.*
 import io.ktor.client.statement.*
+import io.ktor.features.*
 import io.ktor.http.*
 import io.ktor.http.HttpHeaders.Authorization
 import io.ktor.http.auth.*
 import io.ktor.response.*
 import io.ktor.routing.*
+import io.ktor.sessions.*
 import kotlinx.serialization.json.*
 import org.kodein.di.*
+import java.net.URI
+import java.net.URL
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration.Companion.minutes
+import org.kodein.di.ktor.closestDI as di
 
 
+private const val JWT_COOKIE = "jwt"
+private const val SESSION_COOKIE = "drill_session"
+
 class SecurityConfig(override val di: DI) : DIAware {
     private val app by instance<Application>()
     private val authService by instance<UserAuthenticationService>()
-    private val tokenService by instance<TokenService>()
+    private val oauthConfig by instance<OAuthConfig>()
+    private val jwtConfig by instance<JwtConfig>()
 
     init {
         app.install(Authentication) {
 //            configureJwt("jwt")
             configureBasic("basic")
             configureOAuth()
         }
+        app.install(Sessions) {
+            cookie<UserSession>(
+                SESSION_COOKIE,
+                storage = CacheStorage(SessionStorageMemory(), 60.minutes.inWholeMilliseconds)
```

Comment:
CacheStorage ensures that sessions are removed from SessionStorageMemory after timeout
_TODO_ move `60.minutes` to application.conf

Reply:
_TODO_ create task to simplify configuration
- allow use to write `application.conf` directly instead of using env variables
- use kotlin override to customize `application.conf`? (see [1](https://ktor.io/docs/running.html))

Reply:
outdated